llvm/lib/CodeGen/MachineBasicBlock.cpp

```diff
@@ -639,6 +639,13 @@ bool MachineBasicBlock::isLiveIn(MCRegister Reg, LaneBitmask LaneMask) const {
   return I != livein_end() && (I->LaneMask & LaneMask).any();
 }
 
+bool MachineBasicBlock::isLiveOut(MCRegister Reg, LaneBitmask LaneMask) const {
```

Comment:
Annoyingly, this function is slightly wrong. One MIR defect is not all of the ABI details are explicit. There's a special case for return values and exceptions. LiveRegUnits/LivePhysRegs account for these (i.e., see addLiveOuts). I'd like to avoid dealing with this kind of question in this change